src/main/java/core/basesyntax/operationstrategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operationstrategy;
+
+import core.basesyntax.dao.FruitStorageDaoImpl;
+import core.basesyntax.model.TransactionDto;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public Integer doOperation(TransactionDto fruitRecord) {
+        int fruitsQuantityAfterBuy = new FruitStorageDaoImpl().getAll()
+                .get(fruitRecord.getFruit()) - fruitRecord.getFruitAmount();
+        if (fruitsQuantityAfterBuy < 0) {
```

Comment:
you repeat yourself in the previous method

Reply:
If you about validation here, it is needed.  It is from task: "NOTE: if while processing transactions you detect some business logic mistakes (e.g not enough fruits to buy) throw an exception from service or handlers."